src/core/thread/mle.cpp

```diff
@@ -2420,6 +2421,53 @@ otError Mle::SendOrphanAnnounce(void)
     return error;
 }
 
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+otError Mle::SendLinkMetricsManagementResponse(const Ip6::Address &aDestination, LinkMetricsStatus aStatus)
+{
+    otError  error = OT_ERROR_NONE;
+    Message *message;
+    Tlv      tlv;
+    ot::Tlv  statusSubTlv;
+
+    VerifyOrExit((message = NewMleMessage()) != nullptr, error = OT_ERROR_NO_BUFS);
+    SuccessOrExit(error = AppendHeader(*message, kCommandLinkMetricsManagementResponse));
+
+    tlv.SetType(Tlv::kLinkMetricsManagement);
+    statusSubTlv.SetType(kLinkMetricsStatus);
+    statusSubTlv.SetLength(sizeof(aStatus));
+    tlv.SetLength(statusSubTlv.GetSize());
+
+    SuccessOrExit(error = message->AppendBytes(&tlv, sizeof(tlv)));
```

Comment:
We can use `Message::Append()` here and all the uses below.

Anywhere we have a pattern like `AppendByte(&someObject, sizeof(someObject))` we can instead use `Append(someObject)`.